gmp-droid-conv.cpp

```diff
@@ -32,7 +32,7 @@ class ConvertNative: public DroidColourConvert
   GMPErr Convert (GMPVideoHost * host, DroidMediaData * in,
       GMPVideoi420Frame * out)
   {
-    int32_t size = m_width * m_height;
+    int32_t size = m_stride * m_slice_height;
```

Comment:
Just a note that in other places where `CreateFrame()` is used I see `m_stride * m_height` as the size; in case `m_slide_height` should be `m_height` here?

Reply:
No, the rest of them take the crop rectangle into account for the height, so they only copy m_height worth of data per frame. Most of the blob colour converters do the full cropping so they only need the old m_width * m_height, and other uses of it I've found online allocate this too, but for some reason one device needs more.

TBH this blob looks really unused these days, and I'm sorely tempted to drop it completely as every device I've tried can use one of the other converters instead. But technically one device may use an opaque format that can only be converted with its own blob, and it's possible that the blob would be more optimized so I'm not quite convinced to drop it yet. There are ports that still crash even with this buffer size increase. No idea what they're doing.

Reply:
Thanks for the explanation :)

Reply:
The output format yuv format of the Qualcomm hardware decoders is I think composed of 64x64 tiles. So an input frame will necessarily be padded accordingly, and the convertor will likewise expect the output to be similarly padded so it can do a tile at a time with no edge case handling.

I'd be concerned if anything is still using the unpadded dimensions for buffer allocations, as it's almost a guaranteed buffer overrun.

Reply:
Hmm well the converter is set up with a crop rectangle, which suggests that the output buffer would conform to that. I had a bit of a look online for any examples of this thing being used. I couldn't find much except for some old Gonk code, and that seemed to be allocating the buffer using the real width and height which i'd expect to crash on this device.
I don't think it will do any harm to over-allocate here, and certainly much less than under-allocating. But from looking at usage, I'm surprised that the header for it still exists.